library/src/methods/getDefaults/getDefaults.ts

```diff
@@ -22,36 +16,25 @@ import type { DefaultValues } from './types.ts';
  *
  * @returns The default values.
  */
-export function getDefaults<
-  TSchema extends SchemaWithMaybeDefault<
-    BaseSchema | ObjectSchema<ObjectEntries, any> | TupleSchema<TupleItems, any>
-  >
->(schema: TSchema): DefaultValues<TSchema> {
-  // Create defaults variable
-  let defaults: any;
-
+export function getDefaults<TSchema extends SchemaWithMaybeDefault>(
+  schema: TSchema
+): DefaultValues<TSchema> | undefined {
   // If schema contains a default function, set its default value
   if (schema.default !== undefined) {
-    defaults = getDefault(schema);
-
-    // Otherwise, check if schema is of kind object or tuple
-  } else if ('type' in schema) {
-    // If it is an object schema, set object with default value of each entry
-    if (schema.type === 'object') {
-      defaults = {};
-      for (const key in schema.entries) {
-        defaults[key] = getDefaults(schema.entries[key]);
-      }
-
-      // If it is a tuple schema, set array with default value of each item
-    } else if (schema.type === 'tuple') {
-      defaults = [];
-      for (let key = 0; key < schema.items.length; key++) {
-        defaults.push(getDefaults(schema.items[key]));
-      }
-    }
+    return getDefault(schema);
+  }
+  // Otherwise, check if schema is of kind object or tuple
+  // If it is an object schema, set object with default value of each entry
+  if (isObjectSchema(schema)) {
```

Comment:
Please refactor this kind of code with your new util function. I am not sure about the name of the util. Something like `match`, `fit` or `comply` could work. Maybe you have better ideas.

Example:

```ts
if (match('string', schema)) {
  // ...
}
```

Reply:
Updated this with the `hasType` utility. Feel free to rename it and modify the usage code comment as you see fit.